### DISCUSSION TITLE
When will the total lines badges be updated?

### DISCUSSION THREAD

**Discussion by jannistpl** [boyter/scc#435] [Q&A]
Hi everyone!

Just wondering when the badges for the total lines will be updated and when the total lines will be counted again.

Yesterday I merged on master with 1,500 new lines, but the badge hasn't been updated yet.
Do I have to do something myself, do I have to create a workflow or does it happen automatically over time?

Thanks in advance!

**Comment by boyter** [comment]
Usually this is down to github caching the badge. You can find some details here https://github.com/boyter/scc/issues/431

I am still investigating a long term fix for this.

**Comment by SMadani** [comment]
Bump. On my project it hasn't updated ever since I added it, and it's been more than a week!

**Comment by boyter** [comment]
Im open to suggestions...

**Reply by SMadani** [reply]
Thanks. I don't think the issue is GitHub caching it, because I tried the "question mark" trick as suggested. I think it's legitimately the cache used by the backend service not being invalidated. It seems to be calculated once per repo and never touched again. My suggestion is to invalidate every time the commit hash changes, with a rate limit of e.g once per 12 hours per repo.

**Reply by jannistpl** [reply]
I can now say with some certainty that this is not a caching issue from Github. It's been over 6 months since I opened this thread and my badge still hasn't been updated. 2.5k lines are still displayed, while the main file already has 2.9k and the total of the `src` folder has 4.7k.
 If I access the badge directly with the browser, without GitHub in between, the badge is also not up to date either. To me this also looks like a caching issue from the backend service.

**Reply by boyter** [reply]
Ah this could be indeed the issue. 

I have pushed a small fix that should age out anything a day old over time. In addition it should have dumped the current cache, so everything should be refreshed now.

Ill keep an eye on this over the next few days, but having some reporting back over the next few days would help a lot.

**Reply by jannistpl** [reply]
The LOC number for me has now been updated from 2.5k to 6k. (But still waiting for the Github cache)
However, I cannot currently say whether the caching problem on the server has been resolved or whether the LOC was only updated by dumping the cache once. Unfortunately, I don't currently have a project where I can make major changes to test this.

**Reply by boyter** [reply]
No problem. I shall keep an eye on things.
